CMakeLists.txt

```
@@ -257,128 +275,56 @@ endif()
 #   ENSMALLEN_INCLUDE_DIR - include directory for ensmallen
 #   STB_IMAGE_INCLUDE_DIR - include directory for STB image library
 #   MATHJAX_ROOT - root of MathJax installation
-find_package(Armadillo "${ARMADILLO_VERSION}" REQUIRED)
 
+# Remove support for older 32bit system, if we need to have 32bit system
+# Since we are cross compiling, We will download it into the deps folder,
+find_package(Armadillo "${ARMADILLO_VERSION}")
+if(NOT ARMADILLO_FOUND AND (NOT DISABLE_DOWNLOADS))
+  get_deps(http://files.mlpack.org/armadillo-10.3.0.tar.gz armadillo armadillo-10.3.0.tar.gz)
+  set(ARMADILLO_INCLUDE_DIR ${GENERIC_INCLUDE_DIR})
+  find_package(Armadillo REQUIRED)
+endif()
 # Include directories for the previous dependencies.
-set(MLPACK_INCLUDE_DIRS ${MLPACK_INCLUDE_DIRS} ${ARMADILLO_INCLUDE_DIRS})
-set(MLPACK_LIBRARIES ${MLPACK_LIBRARIES} ${ARMADILLO_LIBRARIES})
+set(MLPACK_INCLUDE_DIRS ${MLPACK_INCLUDE_DIRS} ${ARMADILLO_INCLUDE_DIR})
```

```suggestion
set(MLPACK_INCLUDE_DIRS ${MLPACK_INCLUDE_DIRS} ${ARMADILLO_INCLUDE_DIRS})
```
Ah, just like `ARMADILLO_LIBRARY` vs. `ARMADILLO_LIBRARIES`, we might find multiple include directories for Armadillo's dependencies, so we need to use `ARMADILLO_INCLUDE_DIRS` here. :+1: